src/wallet/rpcwallet.cpp

```diff
@@ -2262,38 +2262,107 @@ UniValue getstakesplitthreshold(const UniValue& params, bool fHelp)
     return int(pwalletMain->nStakeSplitThreshold);
 }
 
+UniValue getautocombineinfo(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+            "getautocombineinfo\n"
+            "Returns the autocombinerewards settings\n"
+            "\nResult:\n"
+            "1. enabled   (string) The feature turned \"on\" or \"off\".\n"
+            "2. threshold (numeric) If enabled on, returns autocombine threshold.\n"
+            "3. frequency (variable) If enabled, returns frequency in blocks, or \"nextblock\" if one time. "
+                                    "If one time already run, \"startup\" is returned\n");
+
+    UniValue obj(UniValue::VOBJ);
+    obj.push_back(Pair("enabled", pwalletMain->fCombineDust ? "on" : "off"));
+    if (pwalletMain->fCombineDust) {
+        obj.push_back(Pair("threshold",
+                           int(pwalletMain->nAutoCombineThreshold)));
+        if (pwalletMain->nAutoCombineBlockFrequency == 0) {
+            obj.push_back(Pair("frequency", "nextblock"));
```

Comment:
See above, could use numeric value of `-1`, and add a supplemental return field with the human readable text.

Reply:
Added supplemental field, but rather than putting extra cases and changing cases in the AutoCombineDust() code, frequency of 0 should suffice.  "true" and "0" is "next block", "false" and "0" is "on startup".  JSON response is now numeric, and a supplemental "comment" field is added.